curriculum/challenges/english/25-front-end-development/workshop-email-simulator/6852f19cdc3a4a67ef06f590.md

```diff
@@ -0,0 +1,141 @@
+---
+id: 6852f19cdc3a4a67ef06f590
+title: Step 11
+challengeType: 20
+dashedName: step-11
+---
+
+# --description--
+
+Now you'll add the print statements to display the email information. You'll format the timestamp to show only the date and time in a readable format using the `strftime` method.
+
+The `strftime` method allows you to format datetime objects into strings. The format `'%Y-%m-%d %H:%M'` will show the year, month, day, hour, and minute.
+
+In your `display_full_email` method, after calling `mark_as_read()`, add print statements to display the email header, sender name, receiver name, subject, formatted timestamp, and body. Use f-strings to create a format like this:
+
```

Comment:
I feel this is a bit  too much to do in a single step because people can try to print the email with different approaches. For example, the first thing that comes to my mind is to do something like
```py
def display_full_email(self):
        self.mark_as_read()
        s = f"""
--- Email ---
From: {self.sender}
To: {self.receiver}
Subject: {self.subject}
Received: {self.timestamp.strftime('%Y-%m-%d %H:%M')}
Body: {self.body}
------------
"""
        print(s)
```

Also, they have never used `strftime`, so this might need some explanations.

Reply:
@zairahira 

Was this change applied?